### ISSUE TITLE
ENOTDIR when running a warbled jar with a gem from git

### ISSUE THREAD

**Issue by jeremybmerrill** [jruby/warbler#323]
I'm trying to build a "runnable" script into a jar. The script depends on a gem (`rika`) in a git repo. The jar builds fine (`rake war`), but when I try to run it (`java -jar my-app.jar`), I get:

```
Errno::ENOTDIR: Not a directory - file:/path/to/my/working/dir/my-app.jar!/bundler/gems/rika-8b0d537d8971
```
If I unzip `my-app.jar`, that directory does in fact exist. Strangely enough, a `my-app.jar!` directory is created whenever I run `java -jar my-app.jar`.

I'm using JRuby 1.7.19, Java 8 and Warbler 1.4.7. My config, defined in a Rakefile, is
```
Warbler::Task.new("war",
  Warbler::Config.new { |config|
    config.features = %w(runnable)
    config.jar_name = 'my-app'
    config.jar_extension = 'jar'
    config.dirs = ['lib', 'bin']
    config.override_gem_home = false
  }
)
```
The script is plain vanilla Ruby -- not a Rails app, not a Rack app, etc.

Any help would be appreciated; let me know what I can do to help you.

**Comment by lukehorvat** [comment]
+1. I am encountering the same error.

**Comment by mkristian** [comment]
```file:/path/to/my/working/dir/my-app.jar!/bundler/gems/rika-8b0d537d8971``` is not something java understand but jruby does usually understand it.

there are plenty of fixes regarding those uri-like paths on current jruby-1_7 branch. but I am not sure how easy it is to use a local build with warbler

**Comment by jkutner** [comment]
@jeremybmerrill @lukehorvat Can you create a small project that reproduces this for me? I haven't been able to do so.

**Comment by lukehorvat** [comment]
@jkutner Okay, here you go: https://github.com/lukehorvat/warbler-debug

**Comment by jkutner** [comment]
@lukehorvat thanks! I actually ran into a different error when running `warble runnable war`. I think it because of my version of Bundler. what version are you using?

```
$ warble runnable war
warble aborted!
ArgumentError: dependency name must be a String, was #<Bundler::StubSpecification:0xdc9876b @spec_fetcher=nil, @name="bundler", @source=#<Bundler::Source::Rubygems:0x30b6ffe0 @remotes=[#<URI::HTTPS:0x791d1f8b URL:https://rubygems.org/>], @dependency_names=["sinatra", "jeweler", "pry"], @caches=[<Pathname:/Users/jkutner/workspace/warbler-debug/vendor/cache>, "/Users/jkutner/.rvm/gems/jruby-1.7.19/cache", "/Users/jkutner/.rvm/gems/jruby-1.7.19@global/cache"], @allow_remote=false, @specs=#<Bundler::Index:0x2536 sources=[] specs.size=88>, @installed_specs=#<Bundler::Index:0x2538 sources=[] specs.size=88>, @allow_cached=false, @options={}>, @full_gem_path="/Users/jkutner/.rvm/gems/jruby-1.7.19/gems/bundler-1.10.5", @stub=#<Gem::Specification:0x9ec bundler-1.10.5>, @version=#<Gem::Version "1.10.5">, @platform="ruby">
/Users/jkutner/.rvm/gems/jruby-1.7.19/gems/warbler-1.4.7/lib/warbler/gems.rb:59:in `find_single_gem_files'
/Users/jkutner/.rvm/gems/jruby-1.7.19/gems/warbler-1.4.7/lib/warbler/gems.rb:46:in `specs'
/Users/jkutner/.rvm/gems/jruby-1.7.19/gems/warbler-1.4.7/lib/warbler/gems.rb:46:in `specs'
/Users/jkutner/.rvm/gems/jruby-1.7.19/gems/warbler-1.4.7/lib/warbler/jar.rb:197:in `find_gems_files'
/Users/jkutner/.rvm/gems/jruby-1.7.19/gems/warbler-1.4.7/lib/warbler/jar.rb:144:in `apply'
/Users/jkutner/.rvm/gems/jruby-1.7.19/gems/warbler-1.4.7/lib/warbler/task.rb:118:in `define_files_task'
/Users/jkutner/.rvm/gems/jruby-1.7.19/gems/warbler-1.4.7/lib/warbler/task.rb:94:in `define_main_task'
/Users/jkutner/.rvm/gems/jruby-1.7.19/gems/warbler-1.4.7/lib/warbler/task.rb:93:in `define_main_task'
/Users/jkutner/.rvm/gems/jruby-1.7.19/gems/warbler-1.4.7/lib/warbler/application.rb:77:in `run'
/Users/jkutner/.rvm/gems/jruby-1.7.19/gems/warbler-1.4.7/bin/warble:11:in `(root)'
/Users/jkutner/.rvm/gems/jruby-1.7.19/bin/warble:1:in `(root)'
/Users/jkutner/.rvm/gems/jruby-1.7.19/bin/warble:23:in `(root)'
Tasks: TOP => war:files
(See full trace by running task with --trace)
```

**Comment by lukehorvat** [comment]
bundler 1.10.5, warbler 1.4.8, jruby 1.7.20.

**Comment by jkutner** [comment]
Ah ok. I saw 1.4.7 above. I'm able to reproduce it now

**Comment by jkutner** [comment]
Should be fixed in 1.7.21
